source/particle/interpolator/bilinear_least_squares.cc

```diff
@@ -148,16 +154,88 @@ namespace aspect
-
             // Overshoot and undershoot correction of interpolated particle property.
-            if (interpolated_value > max_value_for_particle_property)
-              interpolated_value = max_value_for_particle_property;
-            else if (interpolated_value < min_value_for_particle_property)
-              interpolated_value = min_value_for_particle_property;
+            if (use_global_valued_limter)
+              {
+                interpolated_value = std::min(interpolated_value, global_maximum_particle_properties[property_index]);
+                interpolated_value = std::max(interpolated_value, global_minimum_particle_properties[property_index]);
+              }
+
+            cell_properties[index_positions][property_index] = interpolated_value;
           }
         return cell_properties;
       }
+
+      template <int dim>
+      void
+      BilinearLeastSquares<dim>::declare_parameters (ParameterHandler &prm)
+      {
+        prm.enter_subsection("Postprocess");
+        {
+          prm.enter_subsection("Particles");
+          {
+            prm.enter_subsection("Interpolator");
+            {
+              prm.enter_subsection("Bilinear least squares");
+              {
+                prm.declare_entry ("Global particle property maximum",
+                                   boost::lexical_cast<std::string>(std::numeric_limits<double>::max()),
+                                   Patterns::List(Patterns::Double ()),
+                                   "The maximum global particle property values that will be used as a "
+                                   "limiter for the bilinear least squares interpolation. The number of the input "
+                                   "'Global particle property maximum' values separated by ',' has to be "
+                                   "the same as the number of the interpolated particle properties.");
+                prm.declare_entry ("Global particle property minimum",
+                                   boost::lexical_cast<std::string>(-std::numeric_limits<double>::max()),
+                                   Patterns::List(Patterns::Double ()),
+                                   "The minimum global particle property that will be used as a "
+                                   "limiter for the bilinear least squares interpolation. The number of the input "
+                                   "'Global particle property minimum' values separated by ',' has to be "
+                                   "the same as the number of the interpolated particle properties.");
+                prm.declare_entry("Use limiter", "false",
+                                  Patterns::Bool (),
+                                  "Whether to apply a global particle property limiting scheme to the interpolated "
+                                  "particle properties using bilinear least squares schmeme. The 'Global particle "
```

Comment:
I would shorten the first sentence as follows: `Whether to apply a global limiting scheme to the interpolated particle properties.` All other information is already available to the user.